hail/src/main/scala/is/hail/expr/ir/EmitFunctionBuilder.scala

```diff
@@ -919,24 +919,37 @@ class EmitMethodBuilder[C](
     }
   }
 
+  private def _invoke[T](_args: Param*): Code[T] = {
+    EmitCodeBuilder.scopedCode(this) { cb =>
```

Comment:
I vote for pushing this up if it isn't hard!

Reply:
It may be hard, but necessary. The reason being, in your later comment you say:

> Let's think about the code we want to generate here. Imagine I have a code parameter p1 and EmitCode parameter p2, where p2 has code tuple types t1 ... tn. I think what we want is:
> 
> ```
> local xm: Boolean = false
> local x1: t1, ..., xn: tn
> if p2.m
>   xm = true
>   (x1, ..., xn) = p2.codeTuple()
> invoke(p1, xm, x1, ..., xn)
> ```
> 
> I don't see how you can do better than that. So I'd flatMap over the (emit)Args to get the args for the non-emit invoke.

We have the idiom for doing this.
```scala
// EmitCodeBuilder
  def memoize[T](ec: EmitCode, name: String): EmitValue
```
The issue is that the CodeBuilder here actually needs to be putting code on the caller, here, we're putting code on the callee.